syntax = "proto3";
package alimama.proto;

message SliceRequest {

  uint64 slice_partition = 1;

  uint64 data_start = 2;

  uint64 data_len = 3;
}

message Request { 
  repeated SliceRequest slice_request = 1; 
}
message Request2 { 
  int32 slice_index = 1;
  repeated SliceRequest slice_request = 2; 
}

message Response {

  int32 status = 1;

  repeated bytes slice_data = 2;
}

message MYRequest {
  int32 CMD = 1;    
  bytes data = 2;   
}

message MYResponse{
  int32 ok = 1;
}


service ModelService {
  rpc Get(Request) returns (Response) {}
  rpc GetSlice(Request2) returns (Response) {}
  rpc CustomCMD(MYRequest) returns (MYResponse) {}
}

